File: plugins/OrchidForms/src/main/java/com/eden/orchid/presentations/FormsModule.java
package com.eden.orchid.presentations;

import com.eden.orchid.OrchidModule;
import com.eden.orchid.api.generators.OrchidGenerator;
import com.eden.orchid.api.options.OptionExtractor;
import com.eden.orchid.api.resources.resourceSource.PluginResourceSource;
import com.eden.orchid.api.theme.components.OrchidComponent;
import com.eden.orchid.presentations.fields.TextField;

public class FormsModule extends OrchidModule {

    @Override
    protected void configure() {
        addToSet(PluginResourceSource.class,
                FormsResourceSource.class);

        addToSet(OrchidGenerator.class,
                FormsGenerator.class);

        addToSet(OrchidComponent.class,
                FormComponent.class);

        addToSet(OptionExtractor.class,
                FormOptionExtractor.class);

        addToSet(FormField.class,
                TextField.class);
    }

}

File: plugins/OrchidForms/src/main/kotlin/com/eden/orchid/forms/FormsModule.kt
package com.eden.orchid.forms

import com.eden.orchid.api.generators.OrchidGenerator
import com.eden.orchid.api.options.OptionExtractor
import com.eden.orchid.api.registration.OrchidModule
import com.eden.orchid.api.theme.components.OrchidComponent
import com.eden.orchid.forms.components.FormComponent
import com.eden.orchid.forms.model.FormField
import com.eden.orchid.forms.model.fields.CheckboxField
import com.eden.orchid.forms.model.fields.DropdownField
import com.eden.orchid.forms.model.fields.HiddenField
import com.eden.orchid.forms.model.fields.RadioButtonsField
import com.eden.orchid.forms.model.fields.TextField
import com.eden.orchid.forms.model.fields.TextareaField
import com.eden.orchid.utilities.addToSet

class FormsModule : OrchidModule() {

    override fun configure() {
        withResources(20)

        addToSet<OrchidGenerator<*>, FormsGenerator>()
        addToSet<OrchidComponent, FormComponent>()
        addToSet<OptionExtractor<*>, FormOptionExtractor>()
        addToSet<FormField>(
                CheckboxField::class,
                RadioButtonsField::class,
                DropdownField::class,
                HiddenField::class,
                TextareaField::class,
                TextField::class)
    }

}
